### PR TITLE
Remove pyspice_circuit data attribute. Closes #67

### LINKED ISSUE #67: Remove unnecessary data attribute in CircuitAnalysisBases (pyspice_circuit)
It might be nice to remove the unnecessary state added by the `self.pyspice_circuit` data attribute. It appears this could simply be passed around as a variable making the functions editing it explicitly acting on it rather than as a side-effect.

### PR DESCRIPTION
**Pull request by brollb**
This moves `self.pyspice_circuit` from a data attribute to be an explicit input/output of the relevant functions.

### CODE REVIEW

File: src/common/plugins/CircuitAnalysisBases.py [hunk 1]
```diff
@@ -535,8 +537,8 @@ class AnalyzeCircuit(CircuitToPySpiceBase):
     """Base class for running various analysis on WebGME node of type Circuit"""
 
     def main(self) -> None:
-        super().convert_to_pyspice(self.active_node)
-        self.run_analytics()
+        circuit = super().convert_to_pyspice(self.active_node)
+        self.run_analytics(circuit)
 
-    def run_analytics(self) -> None:
+    def run_analytics(self, circuit) -> None:
```
Comment (umesh-timalsina) [inline src/common/plugins/CircuitAnalysisBases.py:543]:
```suggestion
    def run_analytics(self, circuit: Union[Circuit, SubCircuit]) -> None:
```

File: src/plugins/RecommendNextComponentsMock/RecommendNextComponentsMock/__init__.py [hunk 2]
```diff
@@ -20,15 +20,15 @@
 class RecommendNextComponentsMock(PluginBase):
     """Runs a mock implementation for recommending components to be added to the Circuit"""
 
-    def run_analytics(self) -> None:
-        self._recommend_next_components()
+    def run_analytics(self, circuit) -> None:
```
Comment (umesh-timalsina) [inline src/plugins/RecommendNextComponentsMock/RecommendNextComponentsMock/__init__.py:23]: Same comments as above.

File: src/plugins/RecommendNextComponentsMock/RecommendNextComponentsMock/__init__.py [hunk 3]
```diff
@@ -20,15 +20,15 @@
...
+        self._recommend_next_components(circuit)
 
-    def _recommend_next_components(self) -> None:
+    def _recommend_next_components(self, circuit) -> None:
```
Comment (umesh-timalsina) [inline src/plugins/RecommendNextComponentsMock/RecommendNextComponentsMock/__init__.py:26]: Same comments as above

**Review by umesh-timalsina — APPROVED** [review]
A simple suggestion on type annotations. Otherwise, LGTM!!

File: src/common/plugins/CircuitAnalysisBases.py [hunk 4]
```diff
@@ -82,8 +86,6 @@ def _initialize(self, circuit: dict) -> None:
         self._ground_pins = set()
         self.nodes_count = 0
         self.pin_labels = dict()
-        circuit_name = self.core.get_attribute(circuit, "name")
-        self.pyspice_circuit = Circuit(circuit_name)
         self._log_debug(f"Initialized empty PySpice circuit for {circuit_name}")
```
Comment (umesh-timalsina) [inline src/common/plugins/CircuitAnalysisBases.py:90]: I think this log information is not necessary now.

Comment (brollb) [reply src/common/plugins/CircuitAnalysisBases.py:90]: ah, fair :)

**Pull request merged**